### ISSUE TITLE
Can't find Docker /Asks to install docker when docker starts after vs code is running.

### ISSUE THREAD

**Issue by jabreland** [microsoft/vscode-remote-release#5460]
Issue Type: <b>Bug</b>

Restart the computer with Docker set to not start on boot. Load VS code where the last workspace was a dev container. Get a messaging informing me that I need docker version 17 installed. Close the dev container window. Start docker desktop and wait until Docker loads. Retry to open the dev container. Get same error about needing to install docker 17.  Close all vs code windows. Right click to get the vs code menu on the main Mac menu. Select to close VS code. Re-open vs code and open a dev container. This time it may work or you may get asked to install docker again. If it didn't work that time then restart VS code one more time and it should work again.

Extension version: 0.191.0
VS Code version: Code 1.59.0 (379476f0e13988d90fab105c5c19e7abc8b1dea8, 2021-08-04T23:14:40.191Z)
OS version: Darwin x64 20.5.0
Restricted Mode: No
Remote OS version: Linux x64 5.10.25-linuxkit

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz (16 x 2300)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|3, 3, 3|
|Memory (System)|16.00GB (0.08GB free)|
|Process Argv|--crash-reporter-id df2f345f-6a2a-4c33-88d3-74aab71d68bf|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|Dev Container: Dark Web|
|OS|Linux x64 5.10.25-linuxkit|
|CPUs|Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz (8 x 2300)|
|Memory (System)|7.77GB (1.20GB free)|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
pythonvspyt602:30300191
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes516:30244333
pythonvspyt639:30300192
pythontb:30283811
pythonptprofiler:30281270
vsdfh931cf:30280410
vshan820:30294714
vstes263cf:30335440
pythondataviewer:30285071
pythonvsuse255:30340121
vscod805:30301674
pythonvspyt200:30340761
vscextlangct:30333562
binariesv615:30325510
pythonvssor306:30344512
bridge0708:30335490
vstre464cf:30350173
bridge0723cf:30351748

```

</details>

**Comment by chrmarti** [comment]
Similar reports in https://github.com/microsoft/vscode-remote-release/issues/5396.

What do you get for `docker version --format {{.Server.APIVersion}}` when Docker is not running?

**Comment by jabreland** [comment]
When docker is not running, I get 
`Error response from daemon: dial unix docker.raw.sock: connect: connection refused`

**Comment by chrmarti** [comment]
Could you check and post what is in the log when you are asked to install Docker (`F1` > `Remote-Containers: Show Container Log`) Make sure to run the latest VS Code version 1.60 and Remote-Containers 0.194.0 or later.

**Comment by jabreland** [comment]
So this is the log when docker is not running: 
```
[2445 ms] Remote-Containers 0.194.0 in VS Code 1.60.1 (83bd43bc519d15e50c4272c6cf5c1479df196a4d).
[2443 ms] Start: Run: git rev-parse --abbrev-ref --symbolic-full-name @{u}
[61744132 ms] Start: Run: docker version --format {{.Server.APIVersion}}
[61744541 ms] 
[61744542 ms] Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
```
However, it worked to start VSCode first then Docker Engine/Desktop this time. It was able to find Docker and then start the container. 
I am wondering if it was the Docker Desktop update I did. It would have been the update with the new licence terms.

**Comment by chrmarti** [comment]
This might have been addressed by the fix in #5396. Marking as duplicate, thanks.
